Hometask/api test junior

KaiZerg opened the pull request:
Add api test junior level

KaiZerg committed:
Add API test junior level

nobugs-me commented on src/test/java/iteration2_junior_level/DepositMoneyTest.java:
```diff
@@ -0,0 +1,173 @@
+                .post("http://localhost:4111/api/v1/accounts")
+                .then()
+                .statusCode(HttpStatus.SC_CREATED);
+    }
```
Где проверка, что аккаунт был создан?

KaiZerg replied:
Сохранил ID созданного аккаунта и добавил проверку, что он появился у клиента через метод GET

nobugs-me commented on src/test/java/iteration2_junior_level/DepositMoneyTest.java:
```diff
@@ -0,0 +1,173 @@
+    }
+
+    @Test
+    public void AuthorizedUserCanDepositMoneyToOwnAccount(){
```
с маленькой буквы методы всегда

nobugs-me commented on src/test/java/iteration2_junior_level/DepositMoneyTest.java:
```diff
@@ -0,0 +1,173 @@
+                .post("http://localhost:4111/api/v1/accounts/deposit") // Используем относительный путь, так как baseURI и port уже установлены
+                .then()
+                .statusCode(HttpStatus.SC_FORBIDDEN)
+                .body(Matchers.equalTo(expectedErrorMessage));
```
Где проверка, что деньги не поступили на счет?

KaiZerg replied:
Добавил проверку, что у нас нет доступа к аккаунту 999 через метод GET

nobugs-me commented on src/test/java/iteration2_junior_level/DepositMoneyTest.java:
```diff
@@ -0,0 +1,173 @@
+                .post("http://localhost:4111/api/v1/accounts/deposit") // Используем относительный путь, так как baseURI и port уже установлены
+                .then()
+                .statusCode(HttpStatus.SC_BAD_REQUEST)
+                .body(Matchers.equalTo(expectedErrorMessage));
```
Где проверка, что деньги не поступили на счет?

KaiZerg replied:
добавил проверку на отсутствие транзакции с amount = 5001.0 напрямую через JsonPath

nobugs-me commented on src/test/java/iteration2_junior_level/TransferMoneyTest.java:
```diff
@@ -0,0 +1,266 @@
+                .post("http://localhost:4111/api/v1/accounts/deposit")
+                .then()
+                .statusCode(HttpStatus.SC_OK);
+
```
Где проверка, что деньги поступили на счет?

nobugs-me commented on src/test/java/iteration2_junior_level/TransferMoneyTest.java:
```diff
@@ -0,0 +1,266 @@
+                .post("http://localhost:4111/api/v1/accounts/transfer") // Используем относительный путь, так как baseURI и port уже установлены
+                .then()
+                .statusCode(HttpStatus.SC_BAD_REQUEST)
+                .body(Matchers.equalTo(expectedErrorMessage));
```
Где проверка, что деньги не поступили на счет?

nobugs-me commented on src/test/java/iteration2_junior_level/TransferMoneyTest.java:
```diff
@@ -0,0 +1,266 @@
+                .post("http://localhost:4111/api/v1/accounts/transfer") // Используем относительный путь, так как baseURI и port уже установлены
+                .then()
+                .statusCode(HttpStatus.SC_BAD_REQUEST)
+                .body(Matchers.equalTo(expectedErrorMessage));
```
Где проверка, что деньги не поступили на счет?

nobugs-me commented on src/test/java/iteration2_junior_level/TransferMoneyTest.java:
```diff
@@ -0,0 +1,266 @@
+                .then()
+                .statusCode(HttpStatus.SC_BAD_REQUEST)
+                .body(Matchers.equalTo(expectedErrorMessage));
+    }
```
Где проверка, что деньги не поступили на счет?

nobugs-me commented on src/test/java/iteration2_junior_level/UpdateCustomerProfileTest.java:
```diff
@@ -0,0 +1,169 @@
+                .then()
+                .statusCode(HttpStatus.SC_OK)
+                .body("customer.name", Matchers.equalTo("John Johnny")) // Проверяем, что имя изменилось
+                .body("message", Matchers.equalTo("Profile updated successfully")); // Дополнительная проверка сообщения
```
Где проверка, что профиль обновился?

nobugs-me reviewed (changes requested):
Привет! 

Я думаю, понятно, что во всех тестах не хватает проверок ожидаемого состояния других ручек, давай исправим.

KaiZerg committed:
Merge branch 'main' into hometask/api-test-junior

KaiZerg committed:
Fix test assertions: correct error messages and JSON path handling

- Update expected error messages in DepositMoneyTest and TransferMoneyTest to match actual API responses
- Fix JsonPath Groovy syntax for negative amount checks
- Fix UpdateCustomerProfileTest: remove message assertion from PUT response (returns empty object) and improve GET response validation
